packages/console/src/components/Checkbox/CheckboxGroup/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { type AdminConsoleKey } from '@logto/phrases';
+import classNames from 'classnames';
+
+import DynamicT from '@/components/DynamicT';
+
+import Checkbox from '../Checkbox';
+
+import * as styles from './index.module.scss';
+
+type Option<T extends string> = {
+  title: AdminConsoleKey;
+  value: T;
+};
+
+type Props<T extends string> = {
+  options: Array<Option<T>>;
+  value: T[];
+  onChange: (value: string[]) => void;
+  className?: string;
+};
+
+function CheckboxGroup<T extends string>({
+  options,
+  value: checkedValues,
+  onChange,
+  className,
+}: Props<T>) {
+  const toggleValue = (value: T) => {
+    onChange(
+      checkedValues.includes(value)
+        ? checkedValues.filter((selected) => selected !== value)
+        : [...checkedValues, value]
+    );
+  };
+
+  return (
+    <div className={classNames(styles.group, className)}>
+      {options.map(({ title, value }) => (
+        <Checkbox
+          key={value}
+          label={<DynamicT forKey={title} />}
+          checked={checkedValues.includes(value)}
+          onChange={() => {
+            toggleValue(value);
+          }}
+        />
```

Comment:
All checkboxes in the same group should share the same `name` prop

Reply:
No `name` prop for the checkbox.

Reply:
Then maybe we should consider adding it, as well as adding an input element back to `Checkbox` component? I removed the input from Checkbox in the last iteration, but now I realized it might be wrong. What do you think?

Because otherwise it will be hard for the react hook form to register it by name.

Reply:
Add an issue to add the input element back: https://linear.app/silverhand/issue/LOG-6066/add-input-element-back-to-checkbox